tasks/logger-file.js

```diff
@@ -0,0 +1,17 @@
+const gulp = require('gulp');
+const pathBuilder = require('../pathBuilder');
+const config = require('../config');
+const file = require('gulp-file');
+const fs = require('fs');
+const gulpif = require('gulp-if');
+
+/**
+ *  `logger-file-create` Task
+ *  ---------------
+ *
+ */
+
+gulp.task('logger:createFile', () => file(config.logger.file, '', { src: true })
+    .pipe(gulpif(!fs.existsSync(`${pathBuilder.jsErrorLoggerSubDir}/${config.logger.file}`),
```

Comment:
In what scenario would the file already exist?

Reply:
If the gulp task runs for the first time then the file would be created, the second time round the file won't be created. The reason the check is there is to prevent gulp from erroring in the scenario where the file might already exist e.g if the task has already run once already inside a project.

Reply:
It's the `sync` bit which concerns me really, feels like we're wedging synchronous code into asynchronous streams. It's probably fine in this instance as it's such a small task 🎲

Reply:
Have done some reading and this is the recommended way accorsing to SO, so I am happy 😸